src/Handlers/PaymentMethodHandler.php

```diff
@@ -55,4 +128,155 @@ public function finalize(
+        }
+        if ($salesChannelContext->getCustomer()->getActiveBillingAddress()->getCountryState()) {
+            $billingState = $salesChannelContext->getCustomer()
+                ->getActiveBillingAddress()->getCountryState()->getShortCode();
+        } else {
+            $billingState = '';
+        }
+
+        //Get customer DOB
+        if ($salesChannelContext->getCustomer()->getBirthday()) {
+            $customerBirthday = $salesChannelContext->getCustomer()->getBirthday()->format('dd-mm-yyyy');
+        } else {
+            $customerBirthday = '';
+        }
+
+        //Validate state.data for payment
+        //TODO replace with stateData stored by generic component branch
+        $stateDataJson = '{
+                              "riskData": {
+                                "clientData": "sample"
+                              },
+                              "paymentMethod": {
+                                "type": "scheme",
+                                "holderName": "Joao Smith",
+                                "encryptedCardNumber": "sample",
+                                "encryptedExpiryMonth": "sample",
+                                "encryptedExpiryYear": "sample",
+                                "encryptedSecurityCode": "sample"
+                              },
+                              "storePaymentMethod": false,
+                              "installments": {
+                                "value": 2
+                              },
+                              "browserInfo": {
+                                "acceptHeader": "*/*",
+                                "colorDepth": 24,
+                                "language": "pt",
+                                "javaEnabled": false,
+                                "screenHeight": 2160,
+                                "screenWidth": 3840,
+                                "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
+                                "timeZoneOffset": -120
+                              }
+                            }';
+
+        //TODO replace arg with $this->paymentStateDataService
+        //->getPaymentStateDataFromContextToken($salesChannelContext->getToken())
+        $stateData = $this->checkoutStateDataValidator->getValidatedAdditionalData(json_decode($stateDataJson, true));
+
+        //Build request
+        $request = array();
+        $request = $this->browserBuilder->buildBrowserData(
+            $_SERVER['HTTP_USER_AGENT'],
+            $_SERVER['HTTP_ACCEPT'],
+            $stateData['browserInfo']['screenWidth'],
+            $stateData['browserInfo']['screenHeight'],
+            $stateData['browserInfo']['colorDepth'],
+            $stateData['browserInfo']['timeZoneOffset'],
+            $stateData['browserInfo']['language'],
+            $stateData['browserInfo']['javaEnabled'],
+            $request
+        );
+        $request = $this->addressBuilder->buildDeliveryAddress(
```

Comment:
'reminder'
What I would do is to get the stateData at first and then check if the stateData has the value you want to include like here: https://github.com/Adyen/adyen-magento2/blob/feature/generic-component/Helper/Requests.php#L120-L148

Reply:
Added a state.data check before assigning the values to the request